modules/gitops-argo-cd-dynamic-scaling.adoc

```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+//
+// * argocd_instance/argo-cd-cr-component-properties.adoc
+
+:_content-type: PROCEDURE
+[id="gitops-argo-cd-dynamic-scaling_{context}"]
+= Enabling dynamic scaling of shards of the Argo CD Application Controller
+
+By default, the Argo CD Application Controller assigns clusters to shards indefinitely. This static assignment can result in uneven distribution of shards when replicas are added or removed. You can enable dynamic scaling of shards to automatically adjust the number of shards based on the number of clusters managed by the Argo CD Application Controller at a given time. This ensures that shards are well balanced and provide for the optimal use of compute resources.
```

Comment:
As I am adding the `round-robin` section, this should be clarified for continuity reasons. But please check with SME if it is correct. Thank you! 
```suggestion
By default, the Argo CD Application Controller assigns clusters to shards indefinitely. If you use the `round-robin` sharding algorithm, this static assignment can result in uneven distribution of shards when replicas are added or removed. You can enable dynamic scaling of shards to automatically adjust the number of shards based on the number of clusters managed by the Argo CD Application Controller at a given time. This ensures that shards are well balanced and provide for the optimal use of compute resources.
```

Reply:
@ekristova Good observation and suggestion!

Reply:
Addressed. Thank you @ekristova.

Reply:
> As I am adding the `round-robin` section, this should be clarified for continuity reasons. But please check with SME if it is correct. Thank you!

Hello @ishitasequeira , I have refined and updated the content, kindly review it. Thank you.